unlock-protocol.com/blog/version-1-2.md

```diff
@@ -0,0 +1,87 @@
+---
+title: "Smart Contract Upgrade: Version 1.2"
+subTitle: We shipped a large upgrade to our Unlock contract and the new locks have new features!
+authorName: Julien Genestoux
+publishDate: January 9, 2020
+description: We wanted to start the year with a big release, so we shipped a very large upgrade (the largest since June 2018) to out smart contracts. This new version brings a ton of new features which will venetually bubble up to the Unlock user interfaces (and beyond!).
+image: /static/images/blog/unlocking-smart-contracts/code.jpeg
+---
+
+We wanted to start the year with a big release, so we shipped a very large upgrade (the largest since June 2018) to out smart contracts. This new version brings a ton of new features which will eventually bubble up to the Unlock user interfaces (and beyond!).
+
+# The Unlock Contract
+
+First, a quick reminder about the architecture behind our smart contracts. First, there is the `Unlock` contract. It's a factory contract, whose main job is to deploy locks for creators. This contract is _currently_ owned by us, at Unlock Inc, through the use of a multi-sig contract (powered by [Gnosis](https://gnosis.io/)), but our hope is that eventually, it can be co-owned by the whole Unlock community.
+
+The `Unlock` contract is upgradable by its owner. This means we can add (and only add!) features to make it more useful. We can also change some of its behavior. For this, we use [ZeppelinOS](https://openzeppelin.com/sdk/) and their proxy approach. The contract at the address `0x3d5409cce1d45233de1d4ebdee74b8e004abdd13` (or `unlock-protocol.eth`) is what is called a _proxy_ contract. Its purpose is to keep track of its internal data (like any smart contract), as well as of the actual _implementation_ which can alter this data. The process to upgrade is fairly strict but guarantees that it will never overwrite any of the past data, while still allow the addition of new features.
```

Comment:
`(and only add!)` technically isn't true... e.g. with this most recent update we actually regressed v0 locks, keys cannot be purchased any longer.  Obviously our goal is to add features but we can remove them too

Reply:
oh that's right. I did not realize that we can actually kill locks.

Reply:
It's not so much that we can kill locks. We are only changing Unlock, and at the moment that's only called by the Lock during a purchase.  When the Lock calls Unlock we can force a revert.  So basically purchase is the only lock feature we could potentially break with an upgrade today.